modules/serdes/include/hephaestus/serdes/protobuf/containers.h

```diff
@@ -0,0 +1,41 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <vector>
+
+#include <google/protobuf/repeated_field.h>
+
+//=================================================================================================
+// NOTE: In order for namespace resolution to work correctly for container serialization introduced
+// in this file, calls to toProto and fromProto must be prefixed with (heph::)serdes::protobuf::
+//=================================================================================================
+
+namespace heph::serdes::protobuf {
+
+//=================================================================================================
+// Vector (std::vector <-> google::protobuf::RepeatedField)
+//=================================================================================================
+template <typename T, typename ProtoT>
+auto toProto(google::protobuf::RepeatedField<ProtoT>& proto_repeated_field, const std::vector<T>& vec)
+    -> void {
+  proto_repeated_field.Clear();  // Ensure that the repeated field is empty before adding elements.
+  proto_repeated_field.Reserve(static_cast<int>(vec.size()));
+  for (const auto& value : vec) {
+    proto_repeated_field.Add(value);
+  }
```

Comment:
how does this even compile? T and ProtoT are different, you are not serializing the vec.
```suggestion
  for (const auto& value : vec) {
    auto *proto_value = proto_repeated_field.Add();   
    toProto(*proto_value, value);
  }
```

Reply:
Good catch. I think it worked as I was testing it on int.